managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java

```diff
@@ -2172,6 +2176,32 @@ public long getCacheSize() {
         return entryCache.getSize();
     }
 
+    /**
+     * return BK error codes that are considered not likely to be recoverable
+     */
+    public static boolean isBkErrorNotRecoverable(int rc) {
+        switch (rc) {
+        case BKException.Code.NoSuchLedgerExistsException:
+        case BKException.Code.ReadException:
```

Comment:
This I'm not 100% sure is non-recoverable, we need to check.

Reply:
yes, I think we can remove it. let me verify in which case we can see `ReadException` and we can remove it if we don't need it.